lib/hal_client/representation_editor.rb

```diff
@@ -100,6 +100,15 @@ def add_link(rel, target, opts={})
       self.class.new(orig_repr, raw.merge("_links" => updated_links_section))
     end
 
+    # Returns a RepresentationEditor exactly like this one except that
+    # is has an new or overwritten property value
+    #
+    # key - The name of the property
+    # value - Value to place in the property
+    def add_property(key, value)
```

Comment:
could we call it `set_property` instead? I think `add_...` might be confusing when one expects the property to be already be set.